File: PrebidMobile/Rendering/EventHandlers/PrebidMobileMoPubAdapters/Sources/PrebidMoPubAdapterConfiguration.m
//
//  PBMMoPubAdapterConfiguration.m
//  PBMMoPubTestAppObjC
//
//  Copyright © 2019 OpenX, Inc. All rights reserved.
//
#import <MoPub.h>

#import <PrebidMobileRendering/PBMSDKConfiguration.h>
#import <PrebidMobileRendering/PBMLogLevel.h>

#import "PrebidMoPubAdapterConfiguration.h"

static NSString * const PREBID_MOPUB_ADAPTER_VERSION = @"0";

@implementation PrebidMoPubAdapterConfiguration

#pragma mark - MPAdapterConfiguration

- (NSString *)adapterVersion {
    return [NSString stringWithFormat:@"%@.%@", PBMSDKConfiguration.sdkVersion, PREBID_MOPUB_ADAPTER_VERSION];
}

- (NSString *)biddingToken {
    return nil;
}

- (NSString *)moPubNetworkName {
    return @"OpenX";
}

- (NSString *)networkSdkVersion {
    return PBMSDKConfiguration.sdkVersion;
}

-(void)initializeNetworkWithConfiguration:(NSDictionary<NSString *,id> *)configuration complete:(void (^)(NSError * _Nullable))complete {
    
    [PBMSDKConfiguration initializeSDK];
    
    PBMSDKConfiguration.singleton.logLevel = PBMLogLevelInfo;
    PBMSDKConfiguration.singleton.locationUpdatesEnabled = YES;
    
    // OpenX's ads include Open Measurement scripts that sometime require additional time for loading.
    PBMSDKConfiguration.singleton.creativeFactoryTimeout = 15;
    
    MPLogInfo(@"Prebid Mobile Rendering SDK initialized succesfully.");
    
    if (complete != nil) {
        complete(nil);
    }
}

@end

File: PrebidMobile/Rendering/EventHandlers/PrebidMobileMoPubAdapters/Sources/PrebidMoPubAdapterConfiguration.swift
//
//  PrebidMoPubAdapterConfiguration.swift
//  PrebidMobileMoPubAdapters
//
//  Copyright © 2021 Prebid. All rights reserved.
//

import Foundation

import MoPubSDK

import PrebidMobileRendering

@objc(PrebidMoPubAdapterConfiguration)
public class PrebidMoPubAdapterConfiguration : MPBaseAdapterConfiguration {
    
    // MARK: - MPAdapterConfiguration
    
    public override var adapterVersion: String {
        "\(PrebidRenderingConfig.shared.version).\(Constants.adapterVersion)"
    }
    
    public override var networkSdkVersion: String {
        PrebidRenderingConfig.shared.version
    }
    
    // NOTE: absence of this property may lead to crash
    public override var moPubNetworkName: String {
        Constants.mopubNetworkName
    }
    
    public override var biddingToken: String? {
        nil
    }
    
    public override func initializeNetwork(withConfiguration configuration: [String : Any]?, complete: ((Error?) -> Void)? = nil) {
        PrebidRenderingConfig.initializeRenderingModule()
        
        PrebidRenderingConfig.shared.logLevel = .info
        PrebidRenderingConfig.shared.locationUpdatesEnabled = true
        PrebidRenderingConfig.shared.creativeFactoryTimeout = 15
        
        complete?(nil)
    }
 }
